core/src/entry.rs

```diff
@@ -233,6 +241,78 @@ impl EntrySlice for [Entry] {
         })
     }
 
+    #[cfg(feature = "cuda")]
+    fn verify(&self, start_hash: &Hash) -> bool {
```

Comment:
This change seems to be causing this issue: https://github.com/solana-labs/solana/issues/4722
I tried using custom thread pool for the par_iters here, but that didn't help.

@TristanDebrunner could you help debug it? Some documentation in this function will also be super helpful.